setup.py

```diff
@@ -0,0 +1,134 @@
+#
+import inspect
+import os
+import subprocess
+import shlex
+
+from setuptools import (
+    setup,
+    Extension,
+    find_packages,
+)
+
+import setuptools.command.build_ext
+
+
+class build_ext(setuptools.command.build_ext.build_ext):
+    """
+    Custom setuptools command class to puppet CMake.
+    """
+
+    def build_extension(self, _ext: Extension):
+        """
+        Hook called by setuptools to build the given Extension.
+
+        @param _ext: Extension to build (we only have one, so ignored).
+        """
+        self.__cmake(
+            [
+                "-S",
+                ".",
+                "-B",
+                self.build_temp,
+                "-G",
+                "Ninja",
+                # Place output artifacts where setuptools expects.
+                "--install-prefix",
+                os.path.abspath(self.build_lib),
+                "--preset",
+                "setuptools",
+            ]
+        )
+
+        self.__cmake(["--build", self.build_temp, "--target", "openassetio-python-module"])
+
+        self.__cmake(
+            [
+                "--install",
+                self.build_temp,
+                "--component",
+                "openassetio-python-module",
+            ]
+        )
+
+    def __cmake(self, args: list[str]):
+        """
+        Execute `cmake` as a subprocess with given arguments.
+
+        @param args: Command-line arguments to pass to `cmake`.
+        """
+        args = ["cmake"] + args
+        self.announce(" ".join(map(shlex.quote, args)), level=2)
+        subprocess.check_call(args, env=os.environ.copy(), stderr=subprocess.STDOUT)
+
+
+setup(
+    name="openassetio",
+    version="1.0.0a4",
+    description=(
+        "An open-source interoperability standard for tools and content management systems used in"
+        " media production."
+    ),
+    long_description=inspect.cleandoc(
+        """ In modern creative pipelines, data is often managed by an authoritative system (Asset
+            Management System, Digital Asset Manager, MAM, et. al).
+
+            It is common for media creation tools to reference this managed data by its present
+            location in a file system.
+
+            OpenAssetIO enables tools to reference managed data by identity (using an "Entity
+            Reference") instead of a file system path.
+
+            This is achieved through the definition of a common set of interactions between a host
+            of the API (eg: a Digital Content Creation tool or pipeline script) and an Asset
+            Management System.
+
+            This common API surface area removes the need for common pipeline business logic to be
+            re-implemented against the native API of each tool, and allows the tools themselves to
+            design new workflows that streamline the creation of complex assets."""
+    ),
+    url="https://github.com/OpenAssetIO/OpenAssetIO",
+    author="The Foundry Visionmongers Ltd",
```

Comment:
I'm no expert on this, but should this be the OpenAssetIO community or something like that since OpenAssetIO is an ASWF project?

Reply:
Good point, I suspect you're correct. I'll have to check on that.